src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -264,6 +264,15 @@ def convert_sp_to_sf_type(datatype: DataType) -> str:
     datetime.time: TimeType,
     bytes: BinaryType,
 }
+if installed_pandas:
+    import numpy
+
+    PYTHON_TO_SNOW_TYPE_MAPPINGS.update(
+        {
+            pandas._libs.tslibs.nattype.NaTType: TimestampType,
```

Comment:
is there a public api form pandas for the `NaTType` or alternatives ways to do this depending on public pandas api?

I raise the question because I recently handled this issue: https://github.com/snowflakedb/snowpark-python/issues/1388.

Reply:
I could make this change, but it seems a little less clean:
```suggestion
            type(pandas.NaT): TimestampType,
```

The main issue is that pandas recommended way of checking for nat is with pandas.isna or pandas.isnull which doesn't fit well with our generic approach here.

Reply:
I personal like this `type(pandas.NaT)` approach, let's make the change?